blog/2025-06-25-introducing-flet-1-0-alpha.md

```diff
@@ -0,0 +1,502 @@
+
+```py
+get_application_cache_directory_async(self) -> str
+get_application_documents_directory_async(self) -> str
+get_application_support_directory_async(self) -> str
+get_downloads_directory_async(self) -> Optional[str]
+get_external_cache_directories_async(self) -> Optional[List[str]]
+get_external_storage_directories_async(self) -> Optional[List[str]]
+get_library_directory_async(self) -> str
+get_external_cache_directory_async(self) -> Optional[str]
+get_temporary_directory_async(self) -> str
+get_console_log_filename_async(self) -> str
+```
+
+🚧 Documentation is in progress 🚧
+
+#### Event handlers without `e`
+
+Event handlers can now omit `e` parameter, for example both of these work:
+
+```py
+button_1.on_click = lambda: print("Clicked!")
+button_2.on_click = lambda e: print("Clicked!", e)
+```
+
+or
+
+```py
+def increment():
+   print("Increment clicked")
+
+inc_btn.on_click = increment
+```
+
+#### `Control.before_event(e)` hook
+
+The method is called before calling any event handler.
+
+It receives an instance of `ControlEvent` as parameter and should return either `True` or `False`. Returning `False` cancels event handler. Example implementation in `Page` class:
+
+```py
+    def before_event(self, e: ControlEvent):
+        if isinstance(e, RouteChangeEvent):
+            if self.route == e.route:
+                return False
+            self.query()
+        return super().before_event(e)
+``` 
+
+#### `ft.context.page` works everywhere
+
+It's now possible to get a reference to a current `Page` instance in any part of Flet program.
+
+## Breaking changes
+
+Flet 1.0 is a major release and includes breaking changes — for good reason!
+
+The Flet team maintains a list of known breaking changes in [this issue](https://github.com/flet-dev/flet/issues/5238).
+
+If you discover something else that’s broken or incorrect, please submit a new issue or discussion. Once confirmed, we’ll update the list.
+
+Below is a summary of the most significant and impactful breaking changes:
+
+### Single-threaded async UI model
+
+Flet 1.0 adopts a single-threaded async UI model, similar to JavaScript or Flutter. This design makes concurrency more predictable and better suited for the browser and mobile platforms.
+
+* Blocking calls like `time.sleep()` will freeze the UI. Instead of `time.sleep()` use `async def` event handlers and using `await asyncio.sleep()` for delays.
+* Switch to async APIs whenever possible.
+* For CPU-bound tasks, offload them to threads using `asyncio.to_thread(...)`.
+
+🚧 Example with CPU-bound method updating progress bar 🚧
+
+### Async control methods
+
+All controls' get- and set- methods are async now.
+
+Methods that do not return any results have fire-and-forget sync wrappers.
+
+🚧 Documentation is in progress 🚧
+
+### `ft.run()` replaces `ft.app()`
+
+Method arguments stay the same. A new `before_main` argument added (see above).
```

Comment:
I don't think the first statement is True, as the first argument in `app()` was named `target` and has now been renamed to `main`.